apps/platform/src/app/projects/project-simulation/project-simulation.component.html

```diff
@@ -1,58 +1,13 @@
-<h1>{{ simulationRun?.name }}</h1>
-
-<div
-  class="simulation"
-  fxLayout="row"
-  fxLayoutAlign="space-between start"
-  fxLayout.lt-md="column"
-  fxLayoutAlign.lt-md=" center space-between"
->
-  <!--    TODO abstract component -->
-  <mat-list role="list">
-    <mat-list-item role="listitem" *ngIf="simulationRun?.id">
-      <biosimulations-icon icon="id" matListIcon> </biosimulations-icon>
-      <h2 matLine>Simulation Id</h2>
-      <p matLine>{{ simulationRun.id }}</p>
-    </mat-list-item>
-    <mat-list-item role="listitem" *ngIf="simulationRun?.simulator">
-      <biosimulations-icon icon="simulator" matListIcon> </biosimulations-icon>
-      <h2 matLine>Simulator</h2>
-      <p matLine>
-        {{ simulationRun.simulator }} : {{ simulationRun.simulatorVersion }}
-      </p>
-    </mat-list-item>
-    <mat-list-item role="listitem" *ngIf="simulationRun?.cpus">
-      <biosimulations-icon icon="processor" matListIcon> </biosimulations-icon>
-      <h2 matLine>CPU</h2>
-      <p matLine>{{ simulationRun.cpus }}</p>
-    </mat-list-item>
-    <mat-list-item role="listitem" *ngIf="simulationRun?.memory">
-      <biosimulations-icon icon="memory" matListIcon> </biosimulations-icon>
-      <h2 matLine>Memory</h2>
-      <p matLine>{{ simulationRun.memory }} GB</p>
-    </mat-list-item>
-
-    <mat-list-item role="listitem" *ngIf="simulationRun?.status">
-      <biosimulations-icon icon="status" matListIcon> </biosimulations-icon>
-      <h2 matLine>Status</h2>
-      <p matLine>{{ simulationRun.status }}</p>
-    </mat-list-item>
-
-    <mat-list-item role="listitem" *ngIf="simulationRun?.submitted">
-      <biosimulations-icon icon="date" matListIcon> </biosimulations-icon>
-      <h2 matLine>Created</h2>
-      <p matLine>{{ simulationRun.submitted | date: 'MMM d, y, h:mm a' }}</p>
-    </mat-list-item>
-
-    <mat-list-item role="listitem" *ngIf="simulationRun?.updated">
-      <biosimulations-icon icon="date" matListIcon> </biosimulations-icon>
-      <h2 matLine>Updated</h2>
-      <p matLine>{{ simulationRun.updated | date: 'MMM d, y, h:mm a' }}</p>
-    </mat-list-item>
-  </mat-list>
-
-  <div>
-    <h2>Logs</h2>
-    <biosimulations-under-construction> </biosimulations-under-construction>
+<ng-container *ngFor="let section of simulationRun">
+  <div *ngIf="section.items.length" class="section">
+    <div class="title">{{ section.title }}</div>
+    <ul class="icon-list">
+      <li *ngFor="let item of section.items">
+        <biosimulations-icon [icon]="item.icon"></biosimulations-icon>
+        <b>{{ item.title }}:</b>{{ ' ' }}
+        <a [href]="item.url" target="_blank" *ngIf="item.url; else noHref">{{ item.value | async }}</a>
```

Comment:
I see now why using the date pipe won't work. You approach is definitely more adaptable here.

Reply:
Another option would be a custom pipe which can format different kinds of data.